### ISSUE TITLE
Hang during triangulation

### ISSUE THREAD

**Issue by mattdev** [andywiecko/BurstTriangulator#31]
Hey,

Firstly, thanks for Burst Triangulator - it's great! Unfortunately I've run into a case, similar to the other thread, where triangulation hangs for certain data. Any help would be very much appreciated!

Here's my input:

float2[] points = {
new float2(31.28938f, 37.67612f), new float2(31.79285f, 37.00624f), new float2(32.03879f, 36.60557f), new float2(32.29923f, 36.36939f), new float2(32.58526f, 36.42342f), new float2(32.876f, 36.53085f), new float2(33.42577f, 36.38619f), new float2(33.88485f, 35.21272f), new float2(34.62434f, 34.02968f), new float2(34.73527f, 33.69278f), new float2(34.86366f, 33.55389f), new float2(35.10379f, 33.08732f), new float2(35.35777f, 32.77784f), new float2(35.69171f, 32.50069f), new float2(35.84656f, 32.22465f), new float2(36.01643f, 32.11908f), new float2(36.17846f, 31.92439f), new float2(36.32175f, 31.51735f), new float2(36.49083f, 31.40269f), new float2(36.6428f, 31.09395f), new float2(36.98143f, 30.87008f), new float2(37.34995f, 30.98518f), new float2(37.65298f, 30.35742f), new float2(38.14125f, 29.79839f), new float2(38.30097f, 29.57764f), new float2(38.63807f, 29.33636f), new float2(38.79191f, 29.04884f), new float2(38.95393f, 28.85409f), new float2(39.28638f, 28.56006f), new float2(39.44593f, 28.33743f), new float2(39.59904f, 28.04167f), new float2(39.76351f, 27.87474f), new float2(40.23344f, 27.10765f), new float2(40.5593f, 26.7389f), new float2(40.64825f, 26.77288f), new float2(40.15792f, 28.3197f), new float2(40.47878f, 27.74801f), new float2(40.6029f, 27.9564f), new float2(40.46587f, 28.58052f), new float2(40.49842f, 29.02806f), new float2(40.85061f, 29.12524f), new float2(41.09647f, 28.99164f), new float2(41.32619f, 28.90017f), new float2(41.62305f, 29.14183f), new float2(41.9069f, 29.41749f), new float2(42.3455f, 29.289f), new float2(42.62182f, 29.07582f), new float2(42.94831f, 28.73164f), new float2(43.10726f, 28.825f), new float2(43.09475f, 29.11192f), new float2(43.03548f, 29.52093f), new float2(43.47666f, 28.87722f), new float2(43.68863f, 28.83212f), new float2(44.24405f, 30.16822f), new float2(44.15243f, 30.4401f), new float2(44.13583f, 30.68668f), new float2(43.97319f, 31.2235f), new float2(43.59188f, 32.07505f), new float2(43.55514f, 32.32842f), new float2(43.28082f, 32.9029f), new float2(43.19778f, 33.17189f), new float2(42.99816f, 33.4802f), new float2(42.90062f, 33.75408f), new float2(42.84206f, 34.01482f), new float2(42.60212f, 34.33672f), new float2(42.51172f, 34.60819f), new float2(42.49178f, 34.8559f), new float2(42.44653f, 35.11214f), new float2(42.40163f, 35.85024f), new float2(42.3139f, 36.1208f), new float2(42.2475f, 36.86615f), new float2(42.09885f, 37.39825f), new float2(41.93857f, 37.69329f), new float2(41.9395f, 38.41592f), new float2(41.74668f, 39.06109f), new float2(41.71388f, 39.41131f), new float2(41.4817f, 39.82877f), new float2(41.40123f, 40.19506f), new float2(41.83162f, 40.72823f), new float2(41.72264f, 41.10414f), new float2(41.74435f, 41.43597f), new float2(41.68886f, 41.79384f), new float2(41.49154f, 42.19954f), new float2(41.38077f, 42.57606f), new float2(41.07294f, 43.35818f), new float2(40.84647f, 43.77372f), new float2(40.94663f, 44.0791f), new float2(40.84524f, 44.45245f), new float2(39.73372f, 45.59874f), new float2(39.7615f, 45.89402f), new float2(39.86456f, 46.3304f), new float2(39.79302f, 46.43954f), new float2(39.95979f, 46.8737f), new float2(40.02057f, 47.10924f), new float2(39.68147f, 46.96014f), new float2(39.87912f, 47.57381f), new float2(39.88906f, 47.83566f), new float2(39.62033f, 47.85281f), new float2(39.28307f, 47.74151f), new float2(38.99869f, 47.72932f), new float2(38.72726f, 47.7414f), new float2(37.57701f, 47.14798f), new float2(37.24718f, 47.0506f), new float2(37.07909f, 47.25636f), new float2(36.70338f, 47.07301f), new float2(36.64832f, 47.4906f), new float2(36.43682f, 47.61499f), new float2(36.1853f, 47.66438f), new float2(35.82184f, 47.50399f), new float2(35.3458f, 47.65338f), new float2(35.14989f, 47.98354f), new float2(35.08235f, 48.16993f), new float2(34.95149f, 48.71061f), new float2(34.80748f, 49.13337f), new float2(35.23618f, 45.39426f), new float2(35.2403f, 45.17966f), new float2(34.40559f, 48.73275f), new float2(34.30258f, 48.75045f), new float2(34.44608f, 47.66825f), new float2(34.40862f, 47.59851f), new float2(34.21555f, 47.40396f), new float2(33.79879f, 47.39106f), new float2(33.84159f, 47.00494f), new float2(33.46966f, 46.95563f), new float2(33.17641f, 47.0181f), new float2(32.9339f, 47.03938f), new float2(32.56659f, 46.98632f), new float2(31.7903f, 47.26541f), new float2(31.56507f, 47.27264f), new float2(31.26646f, 47.33947f), new float2(31.65442f, 46.67304f), new float2(31.69027f, 46.29256f), new float2(32.24787f, 45.48835f), new float2(32.47871f, 44.59815f), new float2(32.53637f, 44.19995f), new float2(32.54522f, 43.84141f), new float2(32.44212f, 43.57378f), new float2(32.21579f, 43.47334f), new float2(31.94443f, 43.40946f), new float2(31.53552f, 43.17303f), new float2(30.83317f, 43.17491f), new float2(30.61916f, 43.06446f), new float2(30.34215f, 43.00517f), new float2(30.04128f, 42.681f), new float2(30.19507f, 41.98758f), new float2(29.75365f, 41.77756f), new float2(29.58416f, 41.34669f), new float2(29.70014f, 40.94119f), new float2(29.99722f, 40.84808f), new float2(30.21854f, 40.72022f), new float2(29.90438f, 40.11747f), new float2(29.82732f, 39.62344f), new float2(30.09426f, 39.28717f), new float2(30.51098f, 39.01957f), new float2(30.38978f, 38.73465f), new float2(30.38398f, 38.04395f), new float2(30.40719f, 37.85884f), new float2(30.80989f, 37.99457f), new float2(31.34938f, 38.09515f)
};
int[] constraintIndices = {
0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10, 11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16, 17, 17, 18, 18, 19, 19, 20, 20, 21, 21, 22, 22, 23, 23, 24, 24, 25, 25, 26, 26, 27, 27, 28, 28, 29, 29, 30, 30, 31, 31, 32, 32, 33, 33, 34, 34, 35, 35, 36, 36, 37, 37, 38, 38, 39, 39, 40, 40, 41, 41, 42, 42, 43, 43, 44, 44, 45, 45, 46, 46, 47, 47, 48, 48, 49, 49, 50, 50, 51, 51, 52, 52, 53, 53, 54, 54, 55, 55, 56, 56, 57, 57, 58, 58, 59, 59, 60, 60, 61, 61, 62, 62, 63, 63, 64, 64, 65, 65, 66, 66, 67, 67, 68, 68, 69, 69, 70, 70, 71, 71, 72, 72, 73, 73, 74, 74, 75, 75, 76, 76, 77, 77, 78, 78, 79, 79, 80, 80, 81, 81, 82, 82, 83, 83, 84, 84, 85, 85, 86, 86, 87, 87, 88, 88, 89, 89, 90, 90, 91, 91, 92, 92, 93, 93, 94, 94, 95, 95, 96, 96, 97, 97, 98, 98, 99, 99, 100, 100, 101, 101, 102, 102, 103, 103, 104, 104, 105, 105, 106, 106, 107, 107, 108, 108, 109, 109, 110, 110, 111, 111, 112, 112, 113, 113, 114, 114, 115, 115, 116, 116, 117, 117, 118, 118, 119, 119, 120, 120, 121, 121, 122, 122, 123, 123, 124, 124, 125, 125, 126, 126, 127, 127, 128, 128, 129, 129, 130, 130, 131, 131, 132, 132, 133, 133, 134, 134, 135, 135, 136, 136, 137, 137, 138, 138, 139, 139, 140, 140, 141, 141, 142, 142, 143, 143, 144, 144, 145, 145, 146, 146, 147, 147, 148, 148, 149, 149, 150, 150, 151, 151, 152, 152, 153, 153, 154, 154, 155, 155, 156, 156, 157, 157, 158, 158, 0
};

using var triangulator = new Triangulator(capacity: 1024, Allocator.Persistent);
using var inputPositions = new NativeArray<float2>(points, Allocator.Persistent);
using var constraintEdges = new NativeArray<int>(constraintIndices, Allocator.Persistent);

var settings = triangulator.Settings;

// Batch count used in parallel job.
settings.BatchCount = 64;
// Triangle is considered as bad if any of its angles is smaller than MinimumAngle. Note: radians.
settings.MinimumAngle = math.radians(33);
// Triangle is not considered as bad if its area is smaller than MinimumArea.
settings.MinimumArea = 0.5f;
// Triangle is considered as bad if its area is greater than MaximumArea.
settings.MaximumArea = 4.0f;
// If true refines mesh using Ruppert's algorithm.
settings.RefineMesh = true;
// If true constrains edges defined in the Triangulator.Input.ConstraintEdges
settings.ConstrainEdges = true;
// If true and provided Triangulator.Input is not valid, it will throw an exception.
settings.ValidateInput = true;
// If true provided point positions and hole seeds are transformed into [-1, 1] box using "center of mass".
settings.UseLocalTransformation = true;
settings.RestoreBoundary = true;

triangulator.Input.Positions = inputPositions;
triangulator.Input.ConstraintEdges = constraintEdges;

triangulator.Run();

Cheers,
Matt

**Comment by andywiecko** [comment]
Hi @mattdev,

Thank you for your feedback! I have just checked your data, and it seems that you have a similar issue to what was reported in issue #30. The _dirty_ solution I provided in that thread should resolve your issue as well. The only difference is that your settings require triangulation with refinement. To achieve this, I added an additional (inverse) transformation to the result.

I was able to obtain the following result:

![image](https://user-images.githubusercontent.com/15237586/229561463-58e90821-fd8c-4a78-81f1-9eac30ef37c5.png)

using this source:

```csharp
using andywiecko.BurstTriangulator;
using System.Linq;
using Unity.Collections;
using Unity.Mathematics;
using UnityEngine;

public class tmp : MonoBehaviour
{
    private void OnDrawGizmos()
    {
        float2[] points = {
            new float2(31.28938f, 37.67612f), new float2(31.79285f, 37.00624f), new float2(32.03879f, 36.60557f), new float2(32.29923f, 36.36939f), new float2(32.58526f, 36.42342f), new float2(32.876f, 36.53085f), new float2(33.42577f, 36.38619f), new float2(33.88485f, 35.21272f), new float2(34.62434f, 34.02968f), new float2(34.73527f, 33.69278f), new float2(34.86366f, 33.55389f), new float2(35.10379f, 33.08732f), new float2(35.35777f, 32.77784f), new float2(35.69171f, 32.50069f), new float2(35.84656f, 32.22465f), new float2(36.01643f, 32.11908f), new float2(36.17846f, 31.92439f), new float2(36.32175f, 31.51735f), new float2(36.49083f, 31.40269f), new float2(36.6428f, 31.09395f), new float2(36.98143f, 30.87008f), new float2(37.34995f, 30.98518f), new float2(37.65298f, 30.35742f), new float2(38.14125f, 29.79839f), new float2(38.30097f, 29.57764f), new float2(38.63807f, 29.33636f), new float2(38.79191f, 29.04884f), new float2(38.95393f, 28.85409f), new float2(39.28638f, 28.56006f), new float2(39.44593f, 28.33743f), new float2(39.59904f, 28.04167f), new float2(39.76351f, 27.87474f), new float2(40.23344f, 27.10765f), new float2(40.5593f, 26.7389f), new float2(40.64825f, 26.77288f), new float2(40.15792f, 28.3197f), new float2(40.47878f, 27.74801f), new float2(40.6029f, 27.9564f), new float2(40.46587f, 28.58052f), new float2(40.49842f, 29.02806f), new float2(40.85061f, 29.12524f), new float2(41.09647f, 28.99164f), new float2(41.32619f, 28.90017f), new float2(41.62305f, 29.14183f), new float2(41.9069f, 29.41749f), new float2(42.3455f, 29.289f), new float2(42.62182f, 29.07582f), new float2(42.94831f, 28.73164f), new float2(43.10726f, 28.825f), new float2(43.09475f, 29.11192f), new float2(43.03548f, 29.52093f), new float2(43.47666f, 28.87722f), new float2(43.68863f, 28.83212f), new float2(44.24405f, 30.16822f), new float2(44.15243f, 30.4401f), new float2(44.13583f, 30.68668f), new float2(43.97319f, 31.2235f), new float2(43.59188f, 32.07505f), new float2(43.55514f, 32.32842f), new float2(43.28082f, 32.9029f), new float2(43.19778f, 33.17189f), new float2(42.99816f, 33.4802f), new float2(42.90062f, 33.75408f), new float2(42.84206f, 34.01482f), new float2(42.60212f, 34.33672f), new float2(42.51172f, 34.60819f), new float2(42.49178f, 34.8559f), new float2(42.44653f, 35.11214f), new float2(42.40163f, 35.85024f), new float2(42.3139f, 36.1208f), new float2(42.2475f, 36.86615f), new float2(42.09885f, 37.39825f), new float2(41.93857f, 37.69329f), new float2(41.9395f, 38.41592f), new float2(41.74668f, 39.06109f), new float2(41.71388f, 39.41131f), new float2(41.4817f, 39.82877f), new float2(41.40123f, 40.19506f), new float2(41.83162f, 40.72823f), new float2(41.72264f, 41.10414f), new float2(41.74435f, 41.43597f), new float2(41.68886f, 41.79384f), new float2(41.49154f, 42.19954f), new float2(41.38077f, 42.57606f), new float2(41.07294f, 43.35818f), new float2(40.84647f, 43.77372f), new float2(40.94663f, 44.0791f), new float2(40.84524f, 44.45245f), new float2(39.73372f, 45.59874f), new float2(39.7615f, 45.89402f), new float2(39.86456f, 46.3304f), new float2(39.79302f, 46.43954f), new float2(39.95979f, 46.8737f), new float2(40.02057f, 47.10924f), new float2(39.68147f, 46.96014f), new float2(39.87912f, 47.57381f), new float2(39.88906f, 47.83566f), new float2(39.62033f, 47.85281f), new float2(39.28307f, 47.74151f), new float2(38.99869f, 47.72932f), new float2(38.72726f, 47.7414f), new float2(37.57701f, 47.14798f), new float2(37.24718f, 47.0506f), new float2(37.07909f, 47.25636f), new float2(36.70338f, 47.07301f), new float2(36.64832f, 47.4906f), new float2(36.43682f, 47.61499f), new float2(36.1853f, 47.66438f), new float2(35.82184f, 47.50399f), new float2(35.3458f, 47.65338f), new float2(35.14989f, 47.98354f), new float2(35.08235f, 48.16993f), new float2(34.95149f, 48.71061f), new float2(34.80748f, 49.13337f), new float2(35.23618f, 45.39426f), new float2(35.2403f, 45.17966f), new float2(34.40559f, 48.73275f), new float2(34.30258f, 48.75045f), new float2(34.44608f, 47.66825f), new float2(34.40862f, 47.59851f), new float2(34.21555f, 47.40396f), new float2(33.79879f, 47.39106f), new float2(33.84159f, 47.00494f), new float2(33.46966f, 46.95563f), new float2(33.17641f, 47.0181f), new float2(32.9339f, 47.03938f), new float2(32.56659f, 46.98632f), new float2(31.7903f, 47.26541f), new float2(31.56507f, 47.27264f), new float2(31.26646f, 47.33947f), new float2(31.65442f, 46.67304f), new float2(31.69027f, 46.29256f), new float2(32.24787f, 45.48835f), new float2(32.47871f, 44.59815f), new float2(32.53637f, 44.19995f), new float2(32.54522f, 43.84141f), new float2(32.44212f, 43.57378f), new float2(32.21579f, 43.47334f), new float2(31.94443f, 43.40946f), new float2(31.53552f, 43.17303f), new float2(30.83317f, 43.17491f), new float2(30.61916f, 43.06446f), new float2(30.34215f, 43.00517f), new float2(30.04128f, 42.681f), new float2(30.19507f, 41.98758f), new float2(29.75365f, 41.77756f), new float2(29.58416f, 41.34669f), new float2(29.70014f, 40.94119f), new float2(29.99722f, 40.84808f), new float2(30.21854f, 40.72022f), new float2(29.90438f, 40.11747f), new float2(29.82732f, 39.62344f), new float2(30.09426f, 39.28717f), new float2(30.51098f, 39.01957f), new float2(30.38978f, 38.73465f), new float2(30.38398f, 38.04395f), new float2(30.40719f, 37.85884f), new float2(30.80989f, 37.99457f), new float2(31.34938f, 38.09515f)
        };
        int[] constraintIndices = {
            0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10, 11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16, 17, 17, 18, 18, 19, 19, 20, 20, 21, 21, 22, 22, 23, 23, 24, 24, 25, 25, 26, 26, 27, 27, 28, 28, 29, 29, 30, 30, 31, 31, 32, 32, 33, 33, 34, 34, 35, 35, 36, 36, 37, 37, 38, 38, 39, 39, 40, 40, 41, 41, 42, 42, 43, 43, 44, 44, 45, 45, 46, 46, 47, 47, 48, 48, 49, 49, 50, 50, 51, 51, 52, 52, 53, 53, 54, 54, 55, 55, 56, 56, 57, 57, 58, 58, 59, 59, 60, 60, 61, 61, 62, 62, 63, 63, 64, 64, 65, 65, 66, 66, 67, 67, 68, 68, 69, 69, 70, 70, 71, 71, 72, 72, 73, 73, 74, 74, 75, 75, 76, 76, 77, 77, 78, 78, 79, 79, 80, 80, 81, 81, 82, 82, 83, 83, 84, 84, 85, 85, 86, 86, 87, 87, 88, 88, 89, 89, 90, 90, 91, 91, 92, 92, 93, 93, 94, 94, 95, 95, 96, 96, 97, 97, 98, 98, 99, 99, 100, 100, 101, 101, 102, 102, 103, 103, 104, 104, 105, 105, 106, 106, 107, 107, 108, 108, 109, 109, 110, 110, 111, 111, 112, 112, 113, 113, 114, 114, 115, 115, 116, 116, 117, 117, 118, 118, 119, 119, 120, 120, 121, 121, 122, 122, 123, 123, 124, 124, 125, 125, 126, 126, 127, 127, 128, 128, 129, 129, 130, 130, 131, 131, 132, 132, 133, 133, 134, 134, 135, 135, 136, 136, 137, 137, 138, 138, 139, 139, 140, 140, 141, 141, 142, 142, 143, 143, 144, 144, 145, 145, 146, 146, 147, 147, 148, 148, 149, 149, 150, 150, 151, 151, 152, 152, 153, 153, 154, 154, 155, 155, 156, 156, 157, 157, 158, 158, 0
        };

        var mu = float2.zero;
        foreach (var p in points)
        {
            mu += p;
        }
        mu /= points.Length;

        var points_mu = points.Select(p => p - mu).ToArray();
        static float2x2 Kron(float2 a, float2 b) => math.float2x2(a * b[0], a * b[1]);

        var cov = float2x2.zero;
        foreach (var p in points_mu)
        {
            cov += Kron(p, p);
        }
        cov /= points_mu.Length;

        Eigen(cov, out _, out var eigvec);

        Gizmos.color = Color.red;
        Gizmos.DrawRay(math.float3(mu, 0), math.float3(eigvec[0], 0));
        Gizmos.color = Color.green;
        Gizmos.DrawRay(math.float3(mu, 0), math.float3(eigvec[1], 0));

        Gizmos.color = Color.white;
        foreach (var p in points)
        {
            Gizmos.DrawSphere(math.float3(p, 0), 0.1f);
        }
        var points_rot = points.Select(p => math.mul(math.transpose(eigvec), p)).ToArray();

        Gizmos.color = Color.yellow;
        foreach (var p in points_rot)
        {
            Gizmos.DrawSphere(math.float3(p, 0), 0.1f);
        }

        float2 min = float.MaxValue;
        float2 max = float.MinValue;
        foreach (var p in points_rot)
        {
            min = math.min(p, min);
            max = math.max(p, max);
        }
        var c = 0.5f * (min + max);
        var s = max - min;

        Gizmos.color = Color.yellow;
        Gizmos.DrawWireCube(math.float3(c, 0), math.float3(s, 0));

        var points_norm = points_rot.Select(p => 2 * (p - c) / s).ToArray();

        Gizmos.color = Color.green;
        foreach (var p in points_norm)
        {
            Gizmos.DrawSphere(math.float3(p, 0), 0.1f);
        }
        Gizmos.DrawWireCube(math.float3(0, 0, 0), math.float3(2, 2, 0));

        using var triangulator = new Triangulator(capacity: 1024, Allocator.Persistent);
        using var inputPositions = new NativeArray<float2>(points_norm, Allocator.Persistent);
        using var constraintEdges = new NativeArray<int>(constraintIndices, Allocator.Persistent);

        triangulator.Settings.ValidateInput = true;
        triangulator.Settings.MinimumArea = 0.01f;
        triangulator.Settings.MaximumArea = 0.1f;
        triangulator.Settings.RefineMesh = true;
        triangulator.Settings.ConstrainEdges = true;
        triangulator.Settings.RestoreBoundary = true;

        triangulator.Input.Positions = inputPositions;
        triangulator.Input.ConstraintEdges = constraintEdges;

        triangulator.Run();

        Gizmos.color = Color.blue;
        var triangles = triangulator.Output.Triangles;
        points = triangulator.Output.Positions.AsArray().Select(p => math.mul(eigvec, p * s / 2f + c)).ToArray();
        for (int i = 0; i < triangles.Length / 3; i++)
        {
            var id0 = triangles[3 * i + 0];
            var id1 = triangles[3 * i + 1];
            var id2 = triangles[3 * i + 2];
            var p0 = math.float3(points[id0], 0);
            var p1 = math.float3(points[id1], 0);
            var p2 = math.float3(points[id2], 0);
            Gizmos.DrawLine(p0, p1);
            Gizmos.DrawLine(p0, p2);
            Gizmos.DrawLine(p1, p2);
        }
    }

    static void Eigen(float2x2 matrix, out float2 eigval, out float2x2 eigvec)
    {
        var a00 = matrix[0][0];
        var a11 = matrix[1][1];
        var a01 = matrix[0][1];

        var a00a11 = a00 - a11;
        var p1 = a00 + a11;
        var p2 = (a00a11 >= 0 ? 1 : -1) * math.sqrt(a00a11 * a00a11 + 4 * a01 * a01);
        var lambda1 = p1 + p2;
        var lambda2 = p1 - p2;
        eigval = 0.5f * math.float2(lambda1, lambda2);

        var phi = 0.5f * math.atan2(2 * a01, a00a11);

        eigvec = math.float2x2
        (
            m00: math.cos(phi), m01: -math.sin(phi),
            m10: math.sin(phi), m11: math.cos(phi)
        );
    }
}
```

As mentioned in issue #30, I plan to fix the editor "hang" issue and add support for "pathological" cases. I hope to have this done by the end of this week or early next week.

Please let me know if you have any further questions or concerns.

Best regards,
Andrzej

**Comment by mattdev** [comment]
That's fantastic, thanks for the speedy solution. I'll keep an eye out for the next release.

**Comment by andywiecko** [comment]
Hey @mattdev ,

I've just released [`v1.5.0`](https://github.com/andywiecko/BurstTriangulator/releases/tag/v1.5.0), which resolves the editor hang during triangulation. 
I have also introduced PCA transformation, which can be enabled with `settings.UsePCATransformation = true;`.
This transformation should address any issues with the triangulation input data that were mentioned in issues #30 and #31. 
I hope that these changes will resolve all of the problems with the project.

I think that this will be the last major version (excluding any necessary bug fixes), and over the next 2-3 months, I plan to prepare version 2.0.0. In this version, I plan to refactor the codebase, remove deprecated features, update packages, and standardize the settings and input format.

If you have any suggestions or feature requests, please let me know so that I can consider additional updates before the release.

Best regards,
Andrzej

---

PS The `1.5.0` is already available on [OpenUPM](https://openupm.com/packages/com.andywiecko.burst.triangulator/).

**Comment by mattdev** [comment]
That's great, thanks @andywiecko. I've just given v1.5.0 a try, and it seems to work correctly with every input I've thrown at it so far :)
